File: Python/FindTheDifference.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        currentLetters = {}
        for i in range(len(s)):
            if s[i] not in currentLetters:
                currentLetters[s[i]] = 1
            else:
                currentLetters[s[i]] += 1
        for i in range(len(t)):
            if t[i] in currentLetters and currentLetters[t[i]] >= 1:
                currentLetters[t[i]] -= 1
            else:
                return t[i]
        # In here, we first make a frequency dictionary by traversing through the list first
        # Then, we traverse the second string, but we subtract if the letter is found in the frequency dictionary
        # If it is not found, OR if the frequency of a letter cannot anymore be subtracted (or it is 0), then that is the difference between the 2 strings

File: C++/FindTheDifference.cpp
class Solution {
public:
    char findTheDifference(string s, string t) {
        map<char, int> letterCountInFirstString;
        // Loop through first string and get the count of each character
        for(int i=0; i<s.size(); i++){
            if(letterCountInFirstString.find(s[i]) == letterCountInFirstString.end()){
                letterCountInFirstString[s[i]] = 1;
            }else{
                letterCountInFirstString[s[i]] += 1;
            }
        }
        // Loop through second string, if found, we subtract by 1, but if already 0, return the character
        // Or return character if it is not found
        for(int i=0; i<t.size(); i++){
            if(letterCountInFirstString.find(t[i]) != letterCountInFirstString.end()){
                if(letterCountInFirstString[t[i]] == 0){
                    return t[i];
                }
                letterCountInFirstString[t[i]] -= 1;
            }
            else{
                return t[i];
            }
        }
        return '\0';
    }
};